root/static/js/cpan.js

```diff
@@ -183,41 +183,81 @@ $(document).ready(function() {
-        preventBadQueries: false,
-        dataType: 'json',
-        lookupLitmit: 20,
-        paramName: 'q',
-        autoSelectFirst: false,
-        // This simply caches the results of a previous search by url (so no reason not to).
-        noCache: false,
-        triggerSelectOnValidInput: false,
-        maxHeight: 180,
-        width: ac_width,
-        transformResult: function(data) {
-            var result = $.map(data, function(row) {
+    search_input.bind('modules_autocomplete', function() {
+        $(this).autocomplete({
+            serviceUrl: '/search/autocomplete',
+            // Wait for more typing rather than firing at every keystroke.
+            deferRequestBy: 150,
+            // If the autocomplete fires with a single colon ("type:") it will get no results
+            // and anything else typed after that will never trigger another query.
+            // Set 'preventBadQueries:false' to keep trying.
+            preventBadQueries: false,
+            dataType: 'json',
+            lookupLitmit: 20,
+            paramName: 'q',
+            autoSelectFirst: false,
+            // This simply caches the results of a previous search by url (so no reason not to).
+            noCache: false,
+            triggerSelectOnValidInput: false,
+            maxHeight: 180,
+            width: ac_width,
+            transformResult: function(data) {
+                var result = $.map(data, function(row) {
+                    return {
+                        data: row.documentation,
+                        value: row.documentation
+                    };
+                });
+                var uniq = {};
+                result = $.grep(result, function(row) {
+                    uniq[row.value] = typeof(uniq[row.value]) == 'undefined' ? 0 : uniq[row.value];
+                    return uniq[row.value]++ < 1;
+                });
                 return {
-                    data: row.documentation,
-                    value: row.documentation
+                    suggestions: result
                 };
-            });
-            var uniq = {};
-            result = $.grep(result, function(row) {
-                uniq[row.value] = typeof(uniq[row.value]) == 'undefined' ? 0 : uniq[row.value];
-                return uniq[row.value]++ < 1;
-            });
-            return {
-                suggestions: result
-            };
-        },
-        onSelect: function(suggestion) {
-            document.location.href = '/pod/' + suggestion.value;
+            },
+            onSelect: function(suggestion) {
+                document.location.href = '/pod/' + suggestion.value;
+            }
+        });
+    });
+
+    search_input.bind('authors_autocomplete', function() {
+        $(this).constructorAutocomplete({
+            key: 'bQssESEa8XUBKzSUjarO',
+            directResults: true,
+            maxHeight: 400,
+            transformResult: function(response) {
```

Comment:
Is the transformResult() function being used, you mean?  Yes, we're doing some trickery to make the PAUSE IDs searchable but not display them as the primary search term.  So we're indexing the string "[PAUSE ID]: [Full name]" so that any match on that string returns results.  But then we're removing everything in front of the colon here so that the full name appears as the primary search term, and the PAUSE ID is the "description".